src/_pytest/warning_types.py

```diff
@@ -55,7 +55,6 @@ class PytestRemovedIn8Warning(PytestDeprecationWarning):
     __module__ = "pytest"
 
 
-@final
 class PytestReturnNotNoneWarning(PytestDeprecationWarning):
```

Comment:
@RonnyPfannschmidt requested that PytestReturnNotNoneWarning be a RemovedIn8Warning and I agree